src/libraries/Hooks.sol

```diff
@@ -125,12 +127,8 @@ library Hooks {
         (success, result) = address(self).call(data);
         if (!success) _revert(result);
 
-        bytes4 expectedSelector;
-        bytes4 selector;
-        assembly {
-            expectedSelector := mload(add(data, 0x20))
-            selector := mload(add(result, 0x20))
-        }
+        bytes4 expectedSelector = data.parseSelector();
+        bytes4 selector = result.parseSelector();
 
         if (selector != expectedSelector) revert InvalidHookResponse();
```

Comment:
slightly less readable but it removes most of the gas that this change added... can i suggest

```
        // Check expected selector and returned selector match
        if (result.parseSelector() != data.parseSelector()) revert InvalidHookResponse();
```